src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTiming.java

```diff
@@ -319,7 +309,7 @@ public static GenericStatus computeChunkStatus2(@Nonnull WorkflowRun run,
         }
 
         // Previous chunk before end. If flow continued beyond this, it didn't fail.
-        return (run.getResult() == Result.UNSTABLE) ? GenericStatus.UNSTABLE : GenericStatus.SUCCESS;
+        return GenericStatus.SUCCESS;
```

Comment:
I don't think I'm ok with this part of the change, since it means that the unstable status would never be shown for individual stages in Blue Ocean unless someone switches to your workaround with `FlowInterruptedException`. The unfortunate thing is that without this change, your workaround doesn't work either, so nobody is happy. The current visualization isn't particularly meaningful, but based on JENKINS-39203 I feel like changing it without improving the default behavior is probably going to cause additional frustration for users. Maybe we could add a system property to control whether to default to showing unstable or success if the overall result is unstable so that you can opt-in and use your workaround, but we don't change things for other users?

I have some ongoing work in this area to try to come up with a fix for JENKINS-39023, so I think I'll build on top of this PR with some additional changes to fix a few other small issues in preparation for that.

Here is an example Pipeline and the visualizations before and after this PR:

```
pipeline {
    agent any
    stages {
        stage('start') {
            steps {
                echo 'start'
            }
        }
        stage('untable') {
            steps {
                script {
                    currentBuild.result = 'UNSTABLE'
                }
            }
        }
        stage('end') {
            steps {
                echo 'end'
            }
        }
    }
}
```

Before: 
<img width="490" alt="Screen Shot 2019-04-11 at 12 05 43" src="https://user-images.githubusercontent.com/1068968/55973857-5be12200-5c54-11e9-9762-55bac5a1cf30.png">

After:
<img width="566" alt="Screen Shot 2019-04-11 at 12 01 32" src="https://user-images.githubusercontent.com/1068968/55973871-613e6c80-5c54-11e9-857c-0be327db4c78.png">